okapi/static/ocgpx/1/ocgpx.xsd

```diff
@@ -0,0 +1,30 @@
+<?xml version="1.0" encoding="utf-8"?>
+<xsd:schema
+    xmlns:xsd="http://www.w3.org/2001/XMLSchema"
+    xmlns="https://opencaching.pl/okapi/static/ocgpx/1"
+    targetNamespace="https://opencaching.pl/okapi/static/ocgpx/1"
+    elementFormDefault="qualified">
+
+    <xsd:annotation>
+        <xsd:documentation>
+            Common Opencaching extensions to the Topografix GPX schema.
+        </xsd:documentation>
+    </xsd:annotation>
+
+    <xsd:element name="cache" type="cache_type" />
+
+    <xsd:complexType name="cache_type">
+        <xsd:sequence>
+            <xsd:element name="logpw" type="xsd:boolean" minOccurs="1" maxOccurs="1" />
```

Comment:
This must be renamed to "req_passwd".

Reply:
BTW - worth considering - perhaps we could replace this be a more generic A-code attribute element, and introduce a new A-code for this purpose?

Reply:
Since A-codes are quite solid and backward compatible, developers could implement this logic by checking if a particular A-code is present. Also, the users of application which already display attributes, would get this information independently displayed, without the need of the developer implementing support for the new A-code.

(I was quite sure we already had an A-code for "requires passoword", but I checked, and it seems we don't.)

Reply:
Some things to consider:

* There is already a way to deliver OC attributes in GPX, which is understood by some applications; it's called `attrs`=`gc_ocde:attrs` in OKAPI.
* The only major OKAPI App - c:geo - already processes the password requirement as indicated by the `req_passwd` flag of services/caches/geocache(s). Don't know how they display it.
* It may be a good idea to show the password requirement in a more prominent way than an attribute. Because if you are in the field and forget to write down the password, you will not be able to log the cache later. OCPL therefore prominently inserst a text into the GPX description text which points out the password reqirement. 
* It would be the first OKAPI-only attribute, because passwords are stored as strings, while OC attributes are stored as boolean flags. So it would need some change in the OKAPI attribute logic, and it could not be _set_ in the same way like other attributes (see the new [services/caches/edit](https://opencaching.pl/okapi/services/caches/edit.html) method).
* The user's requests that lead to this proposal was to transfer the "requires password" flag from www.opencaching.de via GPX to c:geo.

All things considered, I would prefer to go for the quick solution and replicate the `req_passwd` flag from services/caches/geocaches to GPX. Creating a new GPX attributes standard (besides of the already existing GC and OCDE standards) might be persued separately. Will do no harm if the "requires password" information would be available redundantly.

Reply:
> Will do no harm if the "requires password" information would be available redundantly.

Yes, sure.

It's a pity I never found the time to finish the development of A-codes and S-codes. Probably never will... Despite this, my mind keeps finding new applications for them. :) If designed correctly, they would make such an elegant and forward-compatible model for expressing so many things...! ;)